runtime/runtime/src/lib.rs

```diff
@@ -197,23 +161,20 @@ impl Runtime {
             verifier.verify_transaction(transaction)?
         };
         originator.nonce = transaction.body.get_nonce();
-        Self::apply_rent(&originator_id, &mut originator, block_index);
+        let transaction_cost = self.economics_config.transactions_costs.cost(&transaction.body);
+        originator.checked_sub(transaction_cost)?;
```

Comment:
This check could be done in the verifier.

Reply:
There are many economics-related deductions everywhere throughout the code. We cannot fit all of them into the verifier because they are scattered. So instead of having one deduction in the verifier I suggest we have none.